### DISCUSSION TITLE
session_start() issue - conflict in wordpress

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5109] [Q&A Core] [answered]
Hi, 
I am using class2.php in wordpress enviroment, but  it has issue with sessions.
I found solution that fixed it but I don't know if I can do it in e107 part.

instead `session_start() `

I replaced this:
```
		session_start([
			'read_and_close' => true
		]);
```

and the problem is gone.

More:
https://stackoverflow.com/questions/64377032/getting-an-active-php-session-was-detected-critical-warning-in-wordpress

Thanks for answer

**Comment by Deltik** [comment, marked as answer]
Since [e107 v2.3.0](https://github.com/e107inc/e107/releases/tag/v2.3.0), there is the non-blocking database session handler.  If your problem involves blocking, switching to that session handler might help.

I think specifying the `read_and_close` option to `session_start()` makes the `$_SESSION` read-only, so that would be a no-go for anything that actually needs to write to the session.

Sessions are global-state in PHP.  With WordPress and e107 both trying to initiate a session, I'm not surprised there's a conflict.

On the e107 end, if you don't need the session, you could call `e107::getSession()->shutdown()`.  This could have unintended side effects if done too early.  I'd need to see more details of your application flow to offer better advice.

**Reply by Jimmi08** [reply]
Hi, thank you.  I didn't apply this on live site because one of fixes was - ignore it :)  And now I found that I messed it myself when trying to solve something else. Nevermind.

For your information.  I use it on 2 sites. Both:  e107 is in root, Wordpress is installed in subdirectory, class2.php is first file in wp-config.php.  This works without any issues, I can use e107 tables, shortcodes - for example navigation etc.    

Then I messed up.  Just on localhost.  I needed to include some WP file in e107 plugin, but I had problem with paths.  e107 constants have different value in admin area and in frontend in some cases.  So I added wp-load.php to e107_config.php and this caused this issue. And only reason was to have one correct constant, Example:
`require_once(ABSPATH . "/wp-content/plugins/us-core/us-core.php");`

I am sorry, I totally forgot that this was only on develop installation.

**Reply by Jimmi08** [reply]
You solved it

```
if (!defined('e107_INIT'))
{
	require_once ABSPATH . '../class2.php';
	e107::getSession()->shutdown();
}

```

I moved loading wp-load.php to theme.php, it is loaded only on frontend, so e_BASE is able to use.  And e107_INIT is already defined, so sessions will not be closed in e107 part. 

I am aware of how hight experimental this is, but I am short of time and capabilities. This way I can build a footer with WP Bakery and use it as e107 footer. It works, unbelievable. 

Thanks for help.

**Comment by Jimmi08** [comment]
@Deltik So error was caused by my limited knowledge of how Wordpress works.

This is the solution how to add e107 to WordPress:

in function.php 

```
// Hook for loading e107 system
add_action('init', 'load_e107');

function load_e107()
{
    if (!defined('e107_INIT'))
    {
        require_once(__DIR__ . '/../../../../class2.php');
    }

}
```

Yeah, the path could be solved better way, I just don't have more time.

Any code for inserting as WordPress shortcode and using e107 stuff has to be loaded this way:

```
// Add the shortcode after the init action
add_action('after_setup_theme', 'add_custom_shortcode');

function add_custom_shortcode()
{
        add_shortcode('e107_navigation', 'e107_navigation');
}

```

Example of using {NAVIGATION} in WP:

```
// [e107_navigation]
function e107_navigation($atts)
{
   $text = '{NAVIGATION}';
   $result = e107::getParser()->parseTemplate($text, true, true);
    return $result;
}

```
 
You can't use all e107 functionality. For example, comments are not possible because class2.php is not enough.
